src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,123 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { changeTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  temp?: boolean,
+  deleteTodo?: (id: number) => void,
+  updateTodos?: (filter?: boolean, type?: string) => void
+};
+
+export const Item: React.FC<Props> = ({
+  todo, temp, deleteTodo, updateTodos,
+}) => {
+  const [loader, setLoader] = useState(false);
+  const [change, setChange] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+
+  const activeLoader = () => {
+    if (deleteTodo) {
+      deleteTodo(todo.id);
+      setLoader(true);
+    }
+  };
+
+  const changeTodoCompleteHandler = (item: Todo, changeName:boolean) => {
+    const result = item;
+
+    setLoader(true);
+    if (!changeName) {
+      result.completed = !item.completed;
+    } else {
+      result.title = title;
+    }
+
+    if (updateTodos) {
+      changeTodo(result.id, result).then(() => {
+        const typeFilter = localStorage.getItem('filter');
+
+        if (!typeFilter) {
+          updateTodos();
+        } else {
+          updateTodos(false, typeFilter);
+        }
+
+        setLoader(false);
+        setChange(false);
+      });
+    }
+  };
+
+  const changeTitleTodoHandler = (item: Todo, event: any) => {
```

Comment:
we can just get this `todo`(first argument) without passing it as parameter

and don't forget to add some type for event

Reply:
if i get todo not how arguments i can't return old value title if push esc